tests/simple/testdata/comparisons.textproto

```diff
@@ -730,10 +809,90 @@ section {
     name: "ne_mixed_types_error"
     expr: "2u != 2"
     disable_check: true
-    eval_error : {
+    eval_error: {
       errors: { message: "no such overload" }
     }
   }
+  test {
+    name: "ne_proto2"
+    container: "google.api.expr.test.v1.proto2"
+    expr: "TestAllTypes{single_int64: 1234, single_string: '1234'} != TestAllTypes{single_int64: 1234, single_string: '1234'}"
+    value { bool_value: false }
+  }
+  test {
+    name: "ne_proto3"
+    container: "google.api.expr.test.v1.proto3"
+    expr: "TestAllTypes{single_int64: 1234, single_string: '1234'} != TestAllTypes{single_int64: 1234, single_string: '1234'}"
+    value { bool_value: false }
+  }
+  test {
+    name: "ne_proto2_missing_fields_neq"
+    container: "google.api.expr.test.v1.proto2"
+    expr: "TestAllTypes{single_int64: 1234} != TestAllTypes{single_string: '1234'}"
+    value { bool_value: true }
+  }
+  test {
+    name: "ne_proto3_missing_fields_neq"
+    container: "google.api.expr.test.v1.proto3"
+    expr: "TestAllTypes{single_int64: 1234} != TestAllTypes{single_string: '1234'}"
+    value { bool_value: true }
+  }
+  test {
+    name: "ne_proto_nan_equal"
+    container: "google.api.expr.test.v1.proto2"
+    expr: "TestAllTypes{single_double: 2.0 / 0.0} != TestAllTypes{single_double: 1.0 / 0.0}"
+    value { bool_value: false }
+  }
+  test {
+    name: "ne_proto_different_types"
+    container: "google.api.expr.test.v1.proto2"
+    expr: "dyn(TestAllTypes{}) != dyn(NestedTestAllTypes{})"
+    eval_error {
+      errors { message: "no such overload" }
```

Comment:
Similar comment here to as above ... this is the behavior now, but should not be the behavior once heterogeneous equality is supported fully as the types are considered `dyn` at the type-checker